### PR TITLE
Updated references in authorization cheat sheet and typos

### PR DESCRIPTION
**Pull request by joerg-richter-5234**
Reference to Middleware in the Django Framework
The linked page says "This document is for an insecure version of Django that is no longer supported. Please upgrade to a newer release!". Updated the link from 3.1 to 4.0

Reference to OWASP ASVS 4.0, V1.4.1 and V4.1.1
The linked page says "404: Not Found". Updated the link to Version 4.0.3. The bullet points 1.4.1 and 4.1.1 in the document remain valid from my understanding.

Reference to OWASP WSTG 4.5
The linked page says "You're viewing a past version of the Web Security Testing Guide project. For the latest stable version, visit /stable. " Updated the link from Version 4.1 to 4.2

I spotted one or two typos from my understanding, which i hope to squash.

### CODE REVIEW

File: cheatsheets/Authorization_Cheat_Sheet.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Flaws related to authorization logic are a notable concern for web apps. Broken
...
+Both entirely unauthenticated outsiders and authenticated (but not necessarily authorized) users can take advantage of authorization weaknesses.  Although honest mistakes or carelessness on the part of non-malicious entities may enable authorization bypasses, malicious intent is typically required for access control threats to be fully realized.  Horizontal privilege elevation (i.e. being able to access another user's resources) is an especially common weakness that an authenticated user may be able to take advantage of. Faults related to authorization control can allow malicious insiders and outsiders alike to view, modify, or delete sensitive resources of all forms (databases records, static files, personally identifiable information (PII), etc.) or perform actions, such as creating a new account or initiating a costly order, that they should not be privileged to do. Furthermore, if logging related to access control is not properly set-up, such authorization violations may go undetected or a least remain unattributable to a particular individual or group.
```
Comment (joerg-richter-5234) [inline cheatsheets/Authorization_Cheat_Sheet.md:13]: as a non native speaker PII didn't ring a bell, therefore i would like to suggest to add the full term + abbreviation

File: cheatsheets/Authorization_Cheat_Sheet.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Consider the following points and best practices:
...
+Even when no access control rules are explicitly matched, an the application cannot remain neutral when an entity is requesting access to a particular resource. The application must always make a decision, whether implicitly or explicitly, to either deny or permit the requested access. Logic errors and other mistakes relating to access control may happen, especially when access requirements are complex; consequently, one should not rely entirely on explicitly defined rules for matching all possible requests. For security purposes an application should be configured to deny access by default.
```
Comment (joerg-richter-5234) [inline cheatsheets/Authorization_Cheat_Sheet.md:32]: There is a .

File: cheatsheets/Authorization_Cheat_Sheet.md [hunk 3]
```diff
@@ -79,29 +79,29 @@ Although both RBAC and ABAC are popular choices for application access control,
...
 
-- Avoid exposing identifiers to the user when possible. For example it should be possible to retrieve some objects, such as account details,  based solely on currently authenticated user's identity and attributes (e.g. through information contained in a securely implemented JWT or server-side session).
+- Avoid exposing identifiers to the user when possible. For example it should be possible to retrieve some objects, such as account details,  based solely on currently authenticated user's identity and attributes (e.g. through information contained in a securely implemented JSON Web Token (JWT) or server-side session).
```
Comment (joerg-richter-5234) [inline cheatsheets/Authorization_Cheat_Sheet.md:90]: as above: full term + abbreviation

**Review by jmanico — APPROVED** [review]

**Comment by jmanico** [comment]
Thank you for your PR!

**Pull request merged**